tests/unit/test_types.py

```diff
@@ -1048,3 +1048,193 @@ def test_snow_type_to_dtype_str():
 
     with pytest.raises(TypeError, match="invalid DataType"):
         snow_type_to_dtype_str(None)
+
+
+@pytest.mark.parametrize(
+    "tpe, simple_string, json, type_name, json_value",
+    [
+        (DataType(), "data", '"data"', "data", "data"),
+        (BinaryType(), "binary", '"binary"', "binary", "binary"),
+        (BooleanType(), "boolean", '"boolean"', "boolean", "boolean"),
+        (ByteType(), "tinyint", '"byte"', "byte", "byte"),
+        (DateType(), "date", '"date"', "date", "date"),
+        (
+            DecimalType(20, 10),
+            "decimal(20,10)",
+            '"decimal(20,10)"',
+            "decimal",
+            "decimal(20,10)",
+        ),
+        (DoubleType(), "double", '"double"', "double", "double"),
+        (FloatType(), "float", '"float"', "float", "float"),
+        (IntegerType(), "int", '"integer"', "integer", "integer"),
+        (LongType(), "bigint", '"long"', "long", "long"),
+        (ShortType(), "smallint", '"short"', "short", "short"),
```

Comment:
can you look into some other snowflake types as well and see what is appropriate implementation for them. For example, I think Vector type may need some thought. You can create a separate JIRA for them as well if that is more convenient.

Reply:
I will create another Jira and another PR to do it

Reply:
sg!